File: DominioEVendas/src/br/cefetrn/datinf/estoque/dominio/ItemVenda.java
package br.cefetrn.datinf.estoque.dominio;

import java.io.Serializable;

@SuppressWarnings("serial")
//extends produtos  ??tem dependencisomente!
public class ItemVenda /*extends Produto*/ implements Serializable{
	private long id;
	private int qtde;
	private Estado estado;
        private ItemProduto itemProduto;
	
	public Estado getEstado() {
		return estado;
	}
	public void setEstado(Estado estado) {
		this.estado = estado;
	}
	public long getId() {
		return id;
	}
	public void setId(long id) {
		this.id = id;
	}
	public int getQtde() {
		return qtde;
	}
	public void setQtde(int qtde) {
		this.qtde = qtde;
	}

    public ItemProduto getItemProduto() {
        return itemProduto;
    }

    public void setItemProduto(ItemProduto itemProduto) {
        this.itemProduto = itemProduto;
    }
	

   
}

File: eVendas-Site/eVendas/Dominio/ItemVenda.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Dominio
{
    public class ItensVenda : List<ItemVenda>
    {
        public ItemVenda ItemVenda(int IdProduto)
        {
            foreach (ItemVenda itemVenda in this)
            {
                if (itemVenda.ItemProduto.Produto.IdProduto == IdProduto)
                    return itemVenda;
            }
            return null;
        }
    }

    public class ItemVenda
    {
        private int id;

        public int Id
        {
            get { return id; }
            set { id = value; }
        }

        private int qtde;

        public int Qtde
        {
            get { return qtde; }
            set { qtde = value; }
        }

        private Venda venda;

        public Venda Venda
        {
            get { return venda; }
            set { venda = value; }
        }

        private ItemProduto itemProduto;

        public ItemProduto ItemProduto
        {
            get { return itemProduto; }
            set { itemProduto = value; }
        }
	
    }
}
